lib/vmpooler/api/v1.rb

```diff
@@ -109,398 +113,462 @@ def evaluate_template_aliases(template, count)
+          backend.hget("vmpooler__vm__#{vmname}", 'template')
+          jenkins_build_url, user, poolname = backend.exec
+          poolname = poolname.gsub('.', '_')
+
+          if user
+            user = user.gsub('.', '_')
+          else
+            user = 'unauthenticated'
+          end
+          metrics.increment("user.#{user}.#{operation}.#{poolname}")
 
-        url_parts = jenkins_build_url.split('/')[2..-1]
-        jenkins_instance = url_parts[0].gsub('.', '_')
-        value_stream_parts = url_parts[2].split('_')
-        value_stream_parts = value_stream_parts.map { |s| s.gsub('.', '_') }
-        value_stream = value_stream_parts.shift
-        branch = value_stream_parts.pop
-        project = value_stream_parts.shift
-        job_name = value_stream_parts.join('_')
-        build_metadata_parts = url_parts[3]
-        component_to_test = component_to_test('RMM_COMPONENT_TO_TEST_NAME', build_metadata_parts)
+          if jenkins_build_url
+            if jenkins_build_url.include? 'litmus'
+              # Very simple filter for Litmus jobs - just count them coming through for the moment.
+              metrics.increment("usage_litmus.#{user}.#{operation}.#{poolname}")
+              return
+            end
 
-        metrics.increment("usage_jenkins_instance.#{jenkins_instance}.#{value_stream}.#{operation}.#{poolname}")
-        metrics.increment("usage_branch_project.#{branch}.#{project}.#{operation}.#{poolname}")
-        metrics.increment("usage_job_component.#{job_name}.#{component_to_test}.#{operation}.#{poolname}")
+            url_parts = jenkins_build_url.split('/')[2..-1]
+            jenkins_instance = url_parts[0].gsub('.', '_')
+            value_stream_parts = url_parts[2].split('_')
+            value_stream_parts = value_stream_parts.map { |s| s.gsub('.', '_') }
+            value_stream = value_stream_parts.shift
+            branch = value_stream_parts.pop
+            project = value_stream_parts.shift
+            job_name = value_stream_parts.join('_')
+            build_metadata_parts = url_parts[3]
+            component_to_test = component_to_test('RMM_COMPONENT_TO_TEST_NAME', build_metadata_parts)
+
+            metrics.increment("usage_jenkins_instance.#{jenkins_instance}.#{value_stream}.#{operation}.#{poolname}")
+            metrics.increment("usage_branch_project.#{branch}.#{project}.#{operation}.#{poolname}")
+            metrics.increment("usage_job_component.#{job_name}.#{component_to_test}.#{operation}.#{poolname}")
+          end
+        rescue StandardError => e
+          puts 'd', "[!] [#{poolname}] failed while evaluating usage labels on '#{vmname}' with an error: #{e}"
+          span.add_event('error', attributes: {
+            'error.type' => 'Vmpooler::API::V1.update_user_metrics',
+            'error.message' => "[#{poolname}] failed while evaluating usage labels on '#{vmname}' with an error: #{e}"
+          })
+        end
       end
```

Comment:
@sbeaulie Rubocop thows the warning below now that I have added the tracer bit and a `begin` to this method... is there a better way to do this or should I add a control comment to tell Rubocop this is okay?

```ruby
lib/vmpooler/api/v1.rb:301:15: W: Lint/NonLocalExitFromIterator: Non-local exit from iterator, without return value. next, break, Array#find, Array#any?, etc. is preferred.
              return
              ^^^^^^
```